#Como manipular vários formatos de arquivo com Pandas

In [1]:
url = 'https://raw.githubusercontent.com/alura-cursos/Pandas/refs/heads/main/superstore_data.csv'

In [2]:
import pandas as pd

In [3]:
dados_mercado = pd.read_csv(url)

In [4]:
dados_mercado.head()

Id  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0   1826        1970  Graduation       Divorced  84835.0        0         0   
1      1        1961  Graduation         Single  57091.0        0         0   
2  10476        1958  Graduation        Married  67267.0        0         1   
3   1386        1967  Graduation       Together  32474.0        1         1   
4   5371        1989  Graduation         Single  21474.0        1         0   

  Dt_Customer  Recency  MntWines  ...  MntFishProducts  MntSweetProducts  \
0   6/16/2014        0       189  ...              111               189   
1   6/15/2014        0       464  ...                7                 0   
2   5/13/2014        0       134  ...               15                 2   
3   11/5/2014        0        10  ...                0                 0   
4    8/4/2014        0         6  ...               11                 0   

   MntGoldProds  NumDealsPurchases  NumWebPurchases  NumCatalogPurchases  \
0           218                  1                4                    4   
1            37                  1                7                    3   
2            30                  1                3                    2   
3             0                  1                1                    0   
4            34                  2                3                    1   

   NumStorePurchases  NumWebVisitsMonth  Response  Complain  
0                  6                  1         1         0  
1                  7                  5         1         0  
2                  5                  2         0         0  
3                  2                  7         0         0  
4                  2                  7         1         0  

[5 rows x 22 columns]

In [5]:
import chardet

In [6]:
dados_selecao = pd.read_csv(url, usecols=['Id', 'Year_Birth', 'Income'])


In [7]:
dados_selecao.head()

Id  Year_Birth   Income
0   1826        1970  84835.0
1      1        1961  57091.0
2  10476        1958  67267.0
3   1386        1967  32474.0
4   5371        1989  21474.0

In [8]:
dados_selecao = pd.read_csv(url, usecols=[0,1,4])
dados_selecao.head()

Id  Year_Birth   Income
0   1826        1970  84835.0
1      1        1961  57091.0
2  10476        1958  67267.0
3   1386        1967  32474.0
4   5371        1989  21474.0

In [9]:
dados_selecao.to_csv('clientes_merdado.csv', index = False)

In [10]:
clientes_mercado = pd.read_csv('/content/clientes_merdado.csv')

In [11]:
clientes_mercado.head()

Id  Year_Birth   Income
0   1826        1970  84835.0
1      1        1961  57091.0
2  10476        1958  67267.0
3   1386        1967  32474.0
4   5371        1989  21474.0

#Lendo planilhas Excel

In [12]:
#xlsx = formato excel
import pandas as pd
url = 'https://github.com/alura-cursos/Pandas/blob/main/emissoes_CO2.xlsx?raw=True'

In [13]:
dados_co2 = pd.read_excel(url)

In [14]:
dados_co2.head()

País ISO 3166-1 alpha-3   Ano  Total  Carvão  Oléo  Gás  Cimento  \
0  Afeganistão                AFG  1750    0.0     NaN   NaN  NaN      NaN   
1  Afeganistão                AFG  1751    0.0     NaN   NaN  NaN      NaN   
2  Afeganistão                AFG  1752    0.0     NaN   NaN  NaN      NaN   
3  Afeganistão                AFG  1753    0.0     NaN   NaN  NaN      NaN   
4  Afeganistão                AFG  1754    0.0     NaN   NaN  NaN      NaN   

   Queima  Outro  Per Capita  
0     NaN    NaN         NaN  
1     NaN    NaN         NaN  
2     NaN    NaN         NaN  
3     NaN    NaN         NaN  
4     NaN    NaN         NaN

In [15]:
pd.ExcelFile(url).sheet_names

['emissoes_C02', 'emissoes_percapita', 'fontes']

In [16]:
#parâmetos read_excel
dados_co2 = pd.read_excel(url, sheet_name = 'emissoes_C02')
dados_percapita = pd.read_excel(url, sheet_name = 'emissoes_percapita')

In [17]:
dados_percapita.head()

País ISO 3166-1 alpha-3   Ano  Total  Carvão  Óleo  Gás  Cimento  \
0  Afeganistão                AFG  1750    NaN     NaN   NaN  NaN      NaN   
1  Afeganistão                AFG  1751    NaN     NaN   NaN  NaN      NaN   
2  Afeganistão                AFG  1752    NaN     NaN   NaN  NaN      NaN   
3  Afeganistão                AFG  1753    NaN     NaN   NaN  NaN      NaN   
4  Afeganistão                AFG  1754    NaN     NaN   NaN  NaN      NaN   

   Queima  Outro  
0     NaN    NaN  
1     NaN    NaN  
2     NaN    NaN  
3     NaN    NaN  
4     NaN    NaN

In [18]:
fontes = pd.read_excel(url, sheet_name = 'fontes')
fontes.head()

País ISO 3166-1 alpha-3   Ano   Total  Carvão    Oléo     Gás  \
0  Afeganistão                AFG  1750  [NONE]  [NONE]  [NONE]  [NONE]   
1  Afeganistão                AFG  1751  [NONE]  [NONE]  [NONE]  [NONE]   
2  Afeganistão                AFG  1752  [NONE]  [NONE]  [NONE]  [NONE]   
3  Afeganistão                AFG  1753  [NONE]  [NONE]  [NONE]  [NONE]   
4  Afeganistão                AFG  1754  [NONE]  [NONE]  [NONE]  [NONE]   

  Cimento  Queima   Outro Per Capita  
0  [NONE]  [NONE]  [NONE]     [NONE]  
1  [NONE]  [NONE]  [NONE]     [NONE]  
2  [NONE]  [NONE]  [NONE]     [NONE]  
3  [NONE]  [NONE]  [NONE]     [NONE]  
4  [NONE]  [NONE]  [NONE]     [NONE]

In [19]:
intervalo = pd.read_excel(url, sheet_name = 'emissoes_C02', usecols ='A:D')
intervalo.head()

País ISO 3166-1 alpha-3   Ano  Total
0  Afeganistão                AFG  1750    0.0
1  Afeganistão                AFG  1751    0.0
2  Afeganistão                AFG  1752    0.0
3  Afeganistão                AFG  1753    0.0
4  Afeganistão                AFG  1754    0.0

In [20]:
dados_percapita.to_excel('co2_percapita.xlsx', index=False)

In [21]:
pd.read_excel('/content/emissoes_CO2.xlsx')

FileNotFoundError: [Errno 2] No such file or directory: '/content/emissoes_CO2.xlsx'

#Lendo arquivos JSON

In [23]:
import pandas as pd
url = 'https://raw.githubusercontent.com/alura-cursos/Pandas/refs/heads/main/pacientes.json'

In [24]:
pacientes = pd.read_json(url)
pacientes.head()

ID_paciente Doenca_cardiaca    IMC Fumante Consumo_alcool  AVC  \
0            0             Nao  16.60     Sim            Nao  Nao   
1            1             Nao  20.34     Nao            Nao  Sim   
2            2             Nao  26.58     Sim            Nao  Nao   
3            3             Nao  24.21     Nao            Nao  Nao   
4            4             Nao  23.71     Nao            Nao  Nao   

   Saude_fisica  Saude_mental Dificuldade_caminhar Sexo_biologico  \
0             3            30                  Nao       Feminino   
1             0             0                  Nao       Feminino   
2            20            30                  Nao      Masculino   
3             0             0                  Nao       Feminino   
4            28             0                  Sim       Feminino   

  Faixa_etaria    Raca Diabetes Atividade_fisica Saude_geral  Horas_sono Asma  \
0        55-59  Branca      Sim              Sim   Muito boa           5  Sim   
1      80 ou +  Branca      Nao              Sim   Muito boa           7  Nao   
2        65-69  Branca      Sim              Sim    Razoavel           8  Sim   
3        75-79  Branca      Nao               No         Boa           6  Nao   
4        40-44  Branca      Nao              Sim   Muito boa           8  Nao   

  Doenca_renal Cancer_pele  
0          Nao         Sim  
1          Nao         Nao  
2          Nao         Nao  
3          Nao         Sim  
4          Nao         Nao

In [25]:
url2 = 'https://raw.githubusercontent.com/alura-cursos/Pandas/refs/heads/main/pacientes_2.json'

In [26]:
pacientes_2 = pd.read_json(url2)
pacientes_2.head()

Pesquisa   Ano  \
0  Principais Indicadores de Doenca Cardiaca  2020   
1  Principais Indicadores de Doenca Cardiaca  2020   
2  Principais Indicadores de Doenca Cardiaca  2020   

                                           Pacientes  
0  {'ID': '01', 'Faixa_etaria': '55-59', 'Sexo_bi...  
1  {'ID': '02', 'Faixa_etaria': '80 ou +', 'Sexo_...  
2  {'ID': '03', 'Faixa_etaria': '65-69', 'Sexo_bi...

In [27]:
df_normalizado = pd.json_normalize(pacientes_2['Pacientes'])
df_normalizado.head()

ID Faixa_etaria Sexo_biologico    Raça    IMC Fumante Consumo_alcool  \
0  01        55-59       Feminino  Branca  16.60     Sim            Nao   
1  02      80 ou +       Feminino  Branca  20.34     Nao            Nao   
2  03        65-69      Masculino  Branca  26.58     Sim            Nao   

   Saude_fisica  Saude_mental Dificuldade_caminhar Atividade_fisica  \
0             3            30                  Nao              Sim   
1             0             0                  Nao              Sim   
2            20            30                  Nao              Sim   

  Saude_geral  Horas_sono                Problemas_saude  
0   Muito boa           5  [Diabetes, Asma, Cancer_pele]  
1   Muito boa           7                          [AVC]  
2   Muito boa           8               [diabetes, Asma]

In [29]:
#salvando arquivo json
df_normalizado.to_json('pacientes_normalizados.json')
pd.read_json('/content/pacientes_normalizados.json')

ID Faixa_etaria Sexo_biologico    Raça    IMC Fumante Consumo_alcool  \
0   1        55-59       Feminino  Branca  16.60     Sim            Nao   
1   2      80 ou +       Feminino  Branca  20.34     Nao            Nao   
2   3        65-69      Masculino  Branca  26.58     Sim            Nao   

   Saude_fisica  Saude_mental Dificuldade_caminhar Atividade_fisica  \
0             3            30                  Nao              Sim   
1             0             0                  Nao              Sim   
2            20            30                  Nao              Sim   

  Saude_geral  Horas_sono                Problemas_saude  
0   Muito boa           5  [Diabetes, Asma, Cancer_pele]  
1   Muito boa           7                          [AVC]  
2   Muito boa           8               [diabetes, Asma]

In [30]:
import requests
import json

In [31]:
dados_usuarios = requests.get('https://jsonplaceholder.typicode.com/users')
resultado = json.loads(dados_usuarios.text)

In [32]:
import pandas as pd
dados_usuarios_normalizado = pd.json_normalize(resultado, sep='_')

In [33]:
dados_usuarios_normalizado

id                      name          username                      email  \
0   1             Leanne Graham              Bret          Sincere@april.biz   
1   2              Ervin Howell         Antonette          Shanna@melissa.tv   
2   3          Clementine Bauch          Samantha         Nathan@yesenia.net   
3   4          Patricia Lebsack          Karianne  Julianne.OConner@kory.org   
4   5          Chelsey Dietrich            Kamren   Lucio_Hettinger@annie.ca   
5   6      Mrs. Dennis Schulist  Leopoldo_Corkery    Karley_Dach@jasper.info   
6   7           Kurtis Weissnat      Elwyn.Skiles     Telly.Hoeger@billy.biz   
7   8  Nicholas Runolfsdottir V     Maxime_Nienow       Sherwood@rosamond.me   
8   9           Glenna Reichert          Delphine    Chaim_McDermott@dana.io   
9  10        Clementina DuBuque    Moriah.Stanton     Rey.Padberg@karina.biz   

                   phone        website     address_street address_suite  \
0  1-770-736-8031 x56442  hildegard.org        Kulas Light      Apt. 556   
1    010-692-6593 x09125  anastasia.net      Victor Plains     Suite 879   
2         1-463-123-4447    ramiro.info  Douglas Extension     Suite 847   
3      493-170-9623 x156       kale.biz        Hoeger Mall      Apt. 692   
4          (254)954-1289   demarco.info       Skiles Walks     Suite 351   
5   1-477-935-8478 x6430        ola.org  Norberto Crossing      Apt. 950   
6           210.067.6132       elvis.io          Rex Trail     Suite 280   
7      586.493.6943 x140   jacynthe.com   Ellsworth Summit     Suite 729   
8   (775)976-6794 x41206     conrad.com         Dayna Park     Suite 449   
9           024-648-3804    ambrose.net    Kattie Turnpike     Suite 198   

     address_city address_zipcode address_geo_lat address_geo_lng  \
0     Gwenborough      92998-3874        -37.3159         81.1496   
1     Wisokyburgh      90566-7771        -43.9509        -34.4618   
2   McKenziehaven      59590-4157        -68.6102        -47.0653   
3     South Elvis      53919-4257         29.4572       -164.2990   
4      Roscoeview           33263        -31.8129         62.5342   
5   South Christy      23505-1337        -71.4197         71.7478   
6       Howemouth      58804-1099         24.8918         21.8984   
7       Aliyaview           45169        -14.3990       -120.7677   
8  Bartholomebury      76495-3109         24.6463       -168.8889   
9     Lebsackbury      31428-2261        -38.2386         57.2232   

         company_name                       company_catchPhrase  \
0     Romaguera-Crona    Multi-layered client-server neural-net   
1        Deckow-Crist            Proactive didactic contingency   
2  Romaguera-Jacobson         Face to face bifurcated interface   
3       Robel-Corkery  Multi-tiered zero tolerance productivity   
4         Keebler LLC      User-centric fault-tolerant solution   
5   Considine-Lockman        Synchronised bottom-line interface   
6         Johns Group        Configurable multimedia task-force   
7     Abernathy Group             Implemented secondary concept   
8       Yost and Sons     Switchable contextually-based project   
9          Hoeger LLC         Centralized empowering task-force   

                             company_bs  
0           harness real-time e-markets  
1      synergize scalable supply-chains  
2       e-enable strategic applications  
3  transition cutting-edge web services  
4      revolutionize end-to-end systems  
5      e-enable innovative applications  
6         generate enterprise e-tailers  
7         e-enable extensible e-tailers  
8      aggregate real-time technologies  
9              target end-to-end models

#Lendo dados HTML

In [35]:
import pandas as pd
url = 'https://raw.githubusercontent.com/alura-cursos/Pandas/refs/heads/main/filmes_wikipedia.html'

In [38]:
dados_html = pd.read_html(url)
dados_html

[       0                      1
 0   1998             100 Movies
 1   1999              100 Stars
 2   2000             100 Laughs
 3   2001            100 Thrills
 4   2002           100 Passions
 5   2003  100 Heroes & Villains
 6   2004              100 Songs
 7   2005       100 Movie Quotes
 8   2005              25 Scores
 9   2006             100 Cheers
 10  2006            25 Musicals
 11  2007   100 Movies (Updated)
 12  2008        AFI's 10 Top 10
 13   vte                    vte,
                             Film  Release year              Director  \
 0                   Citizen Kane          1941          Orson Welles   
 1                     Casablanca          1942        Michael Curtiz   
 2                  The Godfather          1972  Francis Ford Coppola   
 3             Gone with the Wind          1939        Victor Fleming   
 4             Lawrence of Arabia          1962            David Lean   
 ..                           ...           ...                   

In [39]:
type(dados_html)

list

In [40]:
len(dados_html)

3

In [43]:
top_films = dados_html[1]
top_films

Film  Release year              Director  \
0                   Citizen Kane          1941          Orson Welles   
1                     Casablanca          1942        Michael Curtiz   
2                  The Godfather          1972  Francis Ford Coppola   
3             Gone with the Wind          1939        Victor Fleming   
4             Lawrence of Arabia          1962            David Lean   
..                           ...           ...                   ...   
95                 The Searchers          1956             John Ford   
96              Bringing Up Baby          1938          Howard Hawks   
97                    Unforgiven          1992        Clint Eastwood   
98  Guess Who's Coming to Dinner          1967        Stanley Kramer   
99           Yankee Doodle Dandy          1942        Michael Curtiz   

                      Production companies  Rank  
0                       RKO Radio Pictures     1  
1                    Warner Bros. Pictures     2  
2   Paramount Pictures, Alfran Productions     3  
3          Selznick International Pictures     4  
4                         Horizon Pictures     5  
..                                     ...   ...  
95                  C. V. Whitney Pictures    96  
96                      RKO Radio Pictures    97  
97                     The Malpaso Company    98  
98                       Columbia Pictures    99  
99                   Warner Bros. Pictures   100  

[100 rows x 5 columns]

In [45]:
#Escrevendo arquivos html
top_films.to_html('top_filmes.html')

In [47]:
top_films.to_csv('top_filmes_1998.csv', index=False)

In [48]:
top_films_csv = pd.read_csv('/content/top_filmes_1998.csv')
top_films_csv

Film  Release year              Director  \
0                   Citizen Kane          1941          Orson Welles   
1                     Casablanca          1942        Michael Curtiz   
2                  The Godfather          1972  Francis Ford Coppola   
3             Gone with the Wind          1939        Victor Fleming   
4             Lawrence of Arabia          1962            David Lean   
..                           ...           ...                   ...   
95                 The Searchers          1956             John Ford   
96              Bringing Up Baby          1938          Howard Hawks   
97                    Unforgiven          1992        Clint Eastwood   
98  Guess Who's Coming to Dinner          1967        Stanley Kramer   
99           Yankee Doodle Dandy          1942        Michael Curtiz   

                      Production companies  Rank  
0                       RKO Radio Pictures     1  
1                    Warner Bros. Pictures     2  
2   Paramount Pictures, Alfran Productions     3  
3          Selznick International Pictures     4  
4                         Horizon Pictures     5  
..                                     ...   ...  
95                  C. V. Whitney Pictures    96  
96                      RKO Radio Pictures    97  
97                     The Malpaso Company    98  
98                       Columbia Pictures    99  
99                   Warner Bros. Pictures   100  

[100 rows x 5 columns]

#XML

In [49]:
url = 'https://raw.githubusercontent.com/alura-cursos/Pandas/refs/heads/main/imdb_top_1000.xml'

In [50]:
import pandas as pd

In [52]:
dados_imdb = pd.read_xml(url)
dados_imdb.head(3)

index                                        Poster_Link  \
0      0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1      1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2      2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444

In [53]:
dados_imdb.to_xml('filmes_imdb.xml')

#Banco de dados

In [55]:
import sqlalchemy
from sqlalchemy import create_engine, MetaData, Table, inspect

In [56]:
engine = create_engine('sqlite:///:memory:')

In [57]:
url = 'https://raw.githubusercontent.com/alura-cursos/Pandas/refs/heads/main/clientes_banco.csv'

In [58]:
import pandas as pd

In [59]:
dados = pd.read_csv(url)

In [60]:
dados.head()

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0     5008804     32   Ensino superior  União-estável                2   
1     5008805     32   Ensino superior  União-estável                2   
2     5008806     58      Ensino médio         Casado                2   
3     5008808     52      Ensino médio       Solteiro                1   
4     5008809     52      Ensino médio       Solteiro                1   

    Categoria_de_renda   Ocupacao  Anos_empregado  Rendimento_anual  \
0            Empregado      Outro              12          427500.0   
1            Empregado      Outro              12          427500.0   
2            Empregado  Segurança               3          112500.0   
3  Associado comercial     Vendas               8          270000.0   
4  Associado comercial     Vendas               8          270000.0   

   Tem_carro                   Moradia  
0          1       Apartamento alugado  
1          1       Apartamento alugado  
2          1  Casa/apartamento próprio  
3          0  Casa/apartamento próprio  
4          0  Casa/apartamento próprio

In [61]:
dados.to_sql('clientes', engine, index=False)

438463

In [62]:
inspector = inspect(engine)

In [63]:
print(inspector.get_table_names())

['clientes']


In [66]:
query = 'SELECT * FROM clientes WHERE Categoria_de_renda = "Empregado"'

In [67]:
empregados = pd.read_sql(query, engine)

In [68]:
empregados

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008804     32   Ensino superior  União-estável                2   
1          5008805     32   Ensino superior  União-estável                2   
2          5008806     58      Ensino médio         Casado                2   
3          5008815     46   Ensino superior         Casado                2   
4          5112956     46   Ensino superior         Casado                2   
...            ...    ...               ...            ...              ...   
226054     6837905     43      Ensino médio         Casado                3   
226055     6837906     43      Ensino médio         Casado                3   
226056     6839936     34      Ensino médio         Casado                3   
226057     6840222     43      Ensino médio       Solteiro                1   
226058     6842885     51      Ensino médio         Casado                2   

       Categoria_de_renda          Ocupacao  Anos_empregado  Rendimento_anual  \
0               Empregado             Outro              12          427500.0   
1               Empregado             Outro              12          427500.0   
2               Empregado         Segurança               3          112500.0   
3               Empregado     Contabilidade               2          270000.0   
4               Empregado     Contabilidade               2          270000.0   
...                   ...               ...             ...               ...   
226054          Empregado             Outro               7          355050.0   
226055          Empregado             Outro               7          355050.0   
226056          Empregado  Construção Civil               5          135000.0   
226057          Empregado  Construção Civil               8          103500.0   
226058          Empregado            Vendas               3          121500.0   

        Tem_carro                   Moradia  
0               1       Apartamento alugado  
1               1       Apartamento alugado  
2               1  Casa/apartamento próprio  
3               1  Casa/apartamento próprio  
4               1  Casa/apartamento próprio  
...           ...                       ...  
226054          1  Casa/apartamento próprio  
226055          1  Casa/apartamento próprio  
226056          1  Casa/apartamento próprio  
226057          0  Casa/apartamento próprio  
226058          0  Casa/apartamento próprio  

[226059 rows x 11 columns]

In [69]:
empregados.to_sql('empregados', con=engine, index=False)

226059

In [70]:
pd.read_sql_table('empregados', engine)

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008804     32   Ensino superior  União-estável                2   
1          5008805     32   Ensino superior  União-estável                2   
2          5008806     58      Ensino médio         Casado                2   
3          5008815     46   Ensino superior         Casado                2   
4          5112956     46   Ensino superior         Casado                2   
...            ...    ...               ...            ...              ...   
226054     6837905     43      Ensino médio         Casado                3   
226055     6837906     43      Ensino médio         Casado                3   
226056     6839936     34      Ensino médio         Casado                3   
226057     6840222     43      Ensino médio       Solteiro                1   
226058     6842885     51      Ensino médio         Casado                2   

       Categoria_de_renda          Ocupacao  Anos_empregado  Rendimento_anual  \
0               Empregado             Outro              12          427500.0   
1               Empregado             Outro              12          427500.0   
2               Empregado         Segurança               3          112500.0   
3               Empregado     Contabilidade               2          270000.0   
4               Empregado     Contabilidade               2          270000.0   
...                   ...               ...             ...               ...   
226054          Empregado             Outro               7          355050.0   
226055          Empregado             Outro               7          355050.0   
226056          Empregado  Construção Civil               5          135000.0   
226057          Empregado  Construção Civil               8          103500.0   
226058          Empregado            Vendas               3          121500.0   

        Tem_carro                   Moradia  
0               1       Apartamento alugado  
1               1       Apartamento alugado  
2               1  Casa/apartamento próprio  
3               1  Casa/apartamento próprio  
4               1  Casa/apartamento próprio  
...           ...                       ...  
226054          1  Casa/apartamento próprio  
226055          1  Casa/apartamento próprio  
226056          1  Casa/apartamento próprio  
226057          0  Casa/apartamento próprio  
226058          0  Casa/apartamento próprio  

[226059 rows x 11 columns]

In [72]:
pd.read_sql_table('empregados', engine, columns=['ID_Cliente','Grau_escolaridade', 'Rendimento_anual'])

ID_Cliente Grau_escolaridade  Rendimento_anual
0          5008804   Ensino superior          427500.0
1          5008805   Ensino superior          427500.0
2          5008806      Ensino médio          112500.0
3          5008815   Ensino superior          270000.0
4          5112956   Ensino superior          270000.0
...            ...               ...               ...
226054     6837905      Ensino médio          355050.0
226055     6837906      Ensino médio          355050.0
226056     6839936      Ensino médio          135000.0
226057     6840222      Ensino médio          103500.0
226058     6842885      Ensino médio          121500.0

[226059 rows x 3 columns]

In [73]:
from sqlalchemy import text

In [74]:
query = 'SELECT * FROM clientes'
pd.read_sql(query, engine)

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008804     32   Ensino superior  União-estável                2   
1          5008805     32   Ensino superior  União-estável                2   
2          5008806     58      Ensino médio         Casado                2   
3          5008808     52      Ensino médio       Solteiro                1   
4          5008809     52      Ensino médio       Solteiro                1   
...            ...    ...               ...            ...              ...   
438458     6840104     62      Ensino médio     Divorciado                1   
438459     6840222     43      Ensino médio       Solteiro                1   
438460     6841878     22   Ensino superior       Solteiro                1   
438461     6842765     59      Ensino médio         Casado                2   
438462     6842885     51      Ensino médio         Casado                2   

         Categoria_de_renda          Ocupacao  Anos_empregado  \
0                 Empregado             Outro              12   
1                 Empregado             Outro              12   
2                 Empregado         Segurança               3   
3       Associado comercial            Vendas               8   
4       Associado comercial            Vendas               8   
...                     ...               ...             ...   
438458          Pensionista             Outro               0   
438459            Empregado  Construção Civil               8   
438460  Associado comercial            Vendas               1   
438461          Pensionista             Outro               0   
438462            Empregado            Vendas               3   

        Rendimento_anual  Tem_carro                   Moradia  
0               427500.0          1       Apartamento alugado  
1               427500.0          1       Apartamento alugado  
2               112500.0          1  Casa/apartamento próprio  
3               270000.0          0  Casa/apartamento próprio  
4               270000.0          0  Casa/apartamento próprio  
...                  ...        ...                       ...  
438458          135000.0          0  Casa/apartamento próprio  
438459          103500.0          0  Casa/apartamento próprio  
438460           54000.0          0          Mora com os pais  
438461           72000.0          0  Casa/apartamento próprio  
438462          121500.0          0  Casa/apartamento próprio  

[438463 rows x 11 columns]

In [79]:
query = 'DELETE FROM clientes WHERE ID_Cliente=5008804'
with engine.connect() as conn:
  conn.execute(text(query))
  conn.commit()

In [80]:
pd.read_sql_table('clientes', engine)

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008805     32   Ensino superior  União-estável                2   
1          5008806     58      Ensino médio         Casado                2   
2          5008808     52      Ensino médio       Solteiro                1   
3          5008809     52      Ensino médio       Solteiro                1   
4          5008810     52      Ensino médio       Solteiro                1   
...            ...    ...               ...            ...              ...   
438457     6840104     62      Ensino médio     Divorciado                1   
438458     6840222     43      Ensino médio       Solteiro                1   
438459     6841878     22   Ensino superior       Solteiro                1   
438460     6842765     59      Ensino médio         Casado                2   
438461     6842885     51      Ensino médio         Casado                2   

         Categoria_de_renda          Ocupacao  Anos_empregado  \
0                 Empregado             Outro              12   
1                 Empregado         Segurança               3   
2       Associado comercial            Vendas               8   
3       Associado comercial            Vendas               8   
4       Associado comercial            Vendas               8   
...                     ...               ...             ...   
438457          Pensionista             Outro               0   
438458            Empregado  Construção Civil               8   
438459  Associado comercial            Vendas               1   
438460          Pensionista             Outro               0   
438461            Empregado            Vendas               3   

        Rendimento_anual  Tem_carro                   Moradia  
0               427500.0          1       Apartamento alugado  
1               112500.0          1  Casa/apartamento próprio  
2               270000.0          0  Casa/apartamento próprio  
3               270000.0          0  Casa/apartamento próprio  
4               270000.0          0  Casa/apartamento próprio  
...                  ...        ...                       ...  
438457          135000.0          0  Casa/apartamento próprio  
438458          103500.0          0  Casa/apartamento próprio  
438459           54000.0          0          Mora com os pais  
438460           72000.0          0  Casa/apartamento próprio  
438461          121500.0          0  Casa/apartamento próprio  

[438462 rows x 11 columns]

In [81]:
query = 'UPDATE clientes SET Grau_escolaridade = "Ensino Superior" WHERE ID_Cliente=5008808'

In [82]:
with engine.connect() as conn:
  conn.execute(text(query))
  conn.commit()

In [83]:
pd.read_sql_table('clientes', engine)

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008805     32   Ensino superior  União-estável                2   
1          5008806     58      Ensino médio         Casado                2   
2          5008808     52   Ensino Superior       Solteiro                1   
3          5008809     52      Ensino médio       Solteiro                1   
4          5008810     52      Ensino médio       Solteiro                1   
...            ...    ...               ...            ...              ...   
438457     6840104     62      Ensino médio     Divorciado                1   
438458     6840222     43      Ensino médio       Solteiro                1   
438459     6841878     22   Ensino superior       Solteiro                1   
438460     6842765     59      Ensino médio         Casado                2   
438461     6842885     51      Ensino médio         Casado                2   

         Categoria_de_renda          Ocupacao  Anos_empregado  \
0                 Empregado             Outro              12   
1                 Empregado         Segurança               3   
2       Associado comercial            Vendas               8   
3       Associado comercial            Vendas               8   
4       Associado comercial            Vendas               8   
...                     ...               ...             ...   
438457          Pensionista             Outro               0   
438458            Empregado  Construção Civil               8   
438459  Associado comercial            Vendas               1   
438460          Pensionista             Outro               0   
438461            Empregado            Vendas               3   

        Rendimento_anual  Tem_carro                   Moradia  
0               427500.0          1       Apartamento alugado  
1               112500.0          1  Casa/apartamento próprio  
2               270000.0          0  Casa/apartamento próprio  
3               270000.0          0  Casa/apartamento próprio  
4               270000.0          0  Casa/apartamento próprio  
...                  ...        ...                       ...  
438457          135000.0          0  Casa/apartamento próprio  
438458          103500.0          0  Casa/apartamento próprio  
438459           54000.0          0          Mora com os pais  
438460           72000.0          0  Casa/apartamento próprio  
438461          121500.0          0  Casa/apartamento próprio  

[438462 rows x 11 columns]

Atualizar o registro do cliente de ID 6840104 que teve o rendimento anual alterado para 300000.
Excluir o registro do cliente de ID 5008809, pois essa pessoa não possui mais conta na instituição financeira.
Criar um novo registro de cliente seguindo as especificações abaixo:


  ID_Cliente: 6850985

  Idade: 33
  
  Grau_escolaridade: Doutorado
  
  Estado_civil: Solteiro
  
  Tamanho_familia: 1
  
  Categoria_de_renda: Empregado
  
  Ocupacao: TI
  
  Anos_empregado: 2
  
  Rendimento_anual: 290000
  
  Tem_carro: 0

  Moradia: Casa/apartamento próprio




In [86]:
query = 'UPDATE clientes SET Rendimento_anual = 300000 WHERE ID_Cliente=6840104'

In [88]:
with engine.connect() as conn:
  conn.execute(text(query))
  conn.commit()

In [89]:
pd.read_sql_table('clientes', engine)

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008805     32   Ensino superior  União-estável                2   
1          5008806     58      Ensino médio         Casado                2   
2          5008808     52   Ensino Superior       Solteiro                1   
3          5008809     52      Ensino médio       Solteiro                1   
4          5008810     52      Ensino médio       Solteiro                1   
...            ...    ...               ...            ...              ...   
438457     6840104     62      Ensino médio     Divorciado                1   
438458     6840222     43      Ensino médio       Solteiro                1   
438459     6841878     22   Ensino superior       Solteiro                1   
438460     6842765     59      Ensino médio         Casado                2   
438461     6842885     51      Ensino médio         Casado                2   

         Categoria_de_renda          Ocupacao  Anos_empregado  \
0                 Empregado             Outro              12   
1                 Empregado         Segurança               3   
2       Associado comercial            Vendas               8   
3       Associado comercial            Vendas               8   
4       Associado comercial            Vendas               8   
...                     ...               ...             ...   
438457          Pensionista             Outro               0   
438458            Empregado  Construção Civil               8   
438459  Associado comercial            Vendas               1   
438460          Pensionista             Outro               0   
438461            Empregado            Vendas               3   

        Rendimento_anual  Tem_carro                   Moradia  
0               427500.0          1       Apartamento alugado  
1               112500.0          1  Casa/apartamento próprio  
2               270000.0          0  Casa/apartamento próprio  
3               270000.0          0  Casa/apartamento próprio  
4               270000.0          0  Casa/apartamento próprio  
...                  ...        ...                       ...  
438457          300000.0          0  Casa/apartamento próprio  
438458          103500.0          0  Casa/apartamento próprio  
438459           54000.0          0          Mora com os pais  
438460           72000.0          0  Casa/apartamento próprio  
438461          121500.0          0  Casa/apartamento próprio  

[438462 rows x 11 columns]

In [90]:
query = 'DELETE FROM clientes WHERE ID_cliente=6850985'

In [91]:
with engine.connect() as conn:
  conn.execute(text(query))
  conn.commit()

In [92]:
query = 'INSERT INTO clientes (ID_Cliente, Idade, Grau_escolaridade, Estado_civil, ' \
        'Tamanho_familia, Categoria_de_renda, Ocupacao, Anos_empregado, ' \
        'Rendimento_anual, Tem_carro, Moradia) ' \
        'VALUES (6850985, 33, "Doutorado", "Solteiro", 1, "Empregado", "TI", ' \
        '2, 290000, 0, "Casa/apartamento próprio")'

with engine.connect() as conn:
  result = conn.execute(text(query))
  conn.commit()

In [93]:
pd.read_sql_table('clientes', engine)

ID_Cliente  Idade Grau_escolaridade   Estado_civil  Tamanho_familia  \
0          5008805     32   Ensino superior  União-estável                2   
1          5008806     58      Ensino médio         Casado                2   
2          5008808     52   Ensino Superior       Solteiro                1   
3          5008809     52      Ensino médio       Solteiro                1   
4          5008810     52      Ensino médio       Solteiro                1   
...            ...    ...               ...            ...              ...   
438458     6840222     43      Ensino médio       Solteiro                1   
438459     6841878     22   Ensino superior       Solteiro                1   
438460     6842765     59      Ensino médio         Casado                2   
438461     6842885     51      Ensino médio         Casado                2   
438462     6850985     33         Doutorado       Solteiro                1   

         Categoria_de_renda          Ocupacao  Anos_empregado  \
0                 Empregado             Outro              12   
1                 Empregado         Segurança               3   
2       Associado comercial            Vendas               8   
3       Associado comercial            Vendas               8   
4       Associado comercial            Vendas               8   
...                     ...               ...             ...   
438458            Empregado  Construção Civil               8   
438459  Associado comercial            Vendas               1   
438460          Pensionista             Outro               0   
438461            Empregado            Vendas               3   
438462            Empregado                TI               2   

        Rendimento_anual  Tem_carro                   Moradia  
0               427500.0          1       Apartamento alugado  
1               112500.0          1  Casa/apartamento próprio  
2               270000.0          0  Casa/apartamento próprio  
3               270000.0          0  Casa/apartamento próprio  
4               270000.0          0  Casa/apartamento próprio  
...                  ...        ...                       ...  
438458          103500.0          0  Casa/apartamento próprio  
438459           54000.0          0          Mora com os pais  
438460           72000.0          0  Casa/apartamento próprio  
438461          121500.0          0  Casa/apartamento próprio  
438462          290000.0          0  Casa/apartamento próprio  

[438463 rows x 11 columns]